### ISSUE TITLE
sql: panic in SELECT DISTINCT _, _ FROM (SELECT _, _, _ FROM _._) AS _ JOIN (SELECT _, _, _ FROM _._) AS _ USING (_, _) WHERE ((_ > (-_)) OR ((_ = (-_)) AND (_ > _))) ORDER BY _ LIMIT _

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#21744]
https://sentry.io/cockroach-labs/cockroachdb/issues/439256938/

```
*log.safeError: stopper.go:178: panic while executing 1 statements: SELECT DISTINCT _, _ FROM (SELECT _, _, _ FROM _._) AS _ JOIN (SELECT _, _, _ FROM _._) AS _ USING (_, _) WHERE ((_ > (-_)) OR ((_ = (-_)) AND (_ > _))) ORDER BY _ LIMIT _: caused by <redacted>
  File "github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go", line 429, in func1
  File "github.com/cockroachdb/cockroach/pkg/sql/session.go", line 1477, in maybeRecover
  File "github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go", line 387, in doStop
  File "github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go", line 374, in Stop
  File "github.com/cockroachdb/cockroach/pkg/sql/sem/tree/eval.go", line 2158, in Stop
...
(23 additional frame(s) were not displayed)

stopper.go:178: panic while executing 1 statements: SELECT DISTINCT _, _ FROM (SELECT _, _, _ FROM _._) AS _ JOIN (SELECT _, _, _ FROM _._) AS _ USING (_, _) WHERE ((_ > (-_)) OR ((_ = (-_)) AND (_ > _))) ORDER BY _ LIMIT _: caused by <redacted>
```

**Comment by RaduBerinde** [comment]
@asubiotto could you take a look?

**Comment by arjunravinarayan** [comment]
I did some spelunking to debug this issue without much success. These are some notes of what I tried and why.

A few things: it is pretty clear that this is a memory accounting issue from the sentry.io stacktrace, and this is caused by some memory accounting not being freed at the end of executing a query.

My main suspicion was that the `DISTINCT` operator had changed most recently, and the new pattern of calling might have missed clearing the memory accounting in some edge case. But looking through the code, it checks out, unless I'm missing something very subtle. I could not reproduce this.
